metadata-service/factories/src/main/java/com/linkedin/gms/factory/auth/AuthorizerChainFactory.java

```diff
@@ -65,53 +80,67 @@ protected AuthorizerChain getInstance() {
-    return new AuthorizerContext(resolver);
+  private ResourceSpecResolver initResolver() {
+    return new DefaultResourceSpecResolver(systemAuthentication, entityClient);
   }
 
-  private List<Authorizer> initCustomAuthorizers(AuthorizerContext ctx) {
+  private List<Authorizer> initCustomAuthorizers(ResourceSpecResolver resolver) {
     final List<Authorizer> customAuthorizers = new ArrayList<>();
 
-    if (this.configurationProvider.getAuthorization().getAuthorizers() != null) {
-
-      final List<AuthorizerConfiguration> authorizerConfigurations =
-          this.configurationProvider.getAuthorization().getAuthorizers();
-
-      for (AuthorizerConfiguration authorizer : authorizerConfigurations) {
-        final String type = authorizer.getType();
-        // continue if authorizer is not enabled
-        if (!authorizer.isEnabled()) {
-          log.info(String.format("Authorizer %s is not enabled", type));
-          continue;
-        }
-
-        final Map<String, Object> configs =
-            authorizer.getConfigs() != null ? authorizer.getConfigs() : Collections.emptyMap();
-
-        log.debug(String.format("Found configs for notification sink of type %s: %s ", type, configs));
-
-        // Instantiate the Authorizer
-        Class<? extends Authorizer> clazz = null;
-        try {
-          clazz = (Class<? extends Authorizer>) Class.forName(type);
-        } catch (ClassNotFoundException e) {
-          throw new RuntimeException(
-              String.format("Failed to find custom Authorizer class with name %s on the classpath.", type));
-        }
-
-        // Else construct an instance of the class, each class should have an empty constructor.
-        try {
-          final Authorizer authorizerInstance = clazz.newInstance();
-          authorizerInstance.init(configs, ctx);
-          customAuthorizers.add(authorizerInstance);
-          log.info(String.format("Authorizer %s is initialized", type));
-        } catch (Exception e) {
-          throw new RuntimeException(
-              String.format("Failed to instantiate custom Authorizer with class name %s", clazz.getCanonicalName()), e);
-        }
-      }
-    }
+    Path pluginBaseDirectory = Paths.get(configurationProvider.getDatahub().getPlugin().getAuth().getPath());
+    ConfigProvider configProvider =
+        new ConfigProvider(pluginBaseDirectory);
+
+    Optional<Config> optionalConfig = configProvider.load();
+    // Register authorizer plugins if present
+    optionalConfig.ifPresent((config) -> {
+      registerAuthorizer(customAuthorizers, resolver, config);
+    });
+
     return customAuthorizers;
   }
+
+  private void registerAuthorizer(List<Authorizer> customAuthorizers, ResourceSpecResolver resolver, Config config) {
+    PluginConfigFactory<AuthorizerPluginConfig> authorizerPluginPluginConfigFactory = new PluginConfigFactory<>(config);
```

Comment:
Why do we need the PluginConfigFactory to be typed? 

This abstraction is a bit confusing to me. We already have a Config object. Why cannot we just say 

List<AuthorizerPluginConfig> authorizers = config.loadAuthorizerPluginConfigs()

Reply:
And skip all the factory stuff

Reply:
It is no more typed. if we want to support additional plugins then we just need to add a new config provider in the registry i.e.CONFIG_PROVIDER_REGISTRY